### ISSUE TITLE
certbot prompts re-installation for expired certs

### ISSUE THREAD

**Issue by SwartzCr** [certbot/certbot#3048]
On my server I have an expired cert, but when certbot tries to parse the config file it doesn't see that it needs to be renewed and offers to just re-install it. This might be because we've added a renewal window section to the config file but old files don't have it?
config:

```
cert = /etc/letsencrypt/live/catplanet.cat/cert.pem
privkey = /etc/letsencrypt/live/catplanet.cat/privkey.pem
chain = /etc/letsencrypt/live/catplanet.cat/chain.pem
fullchain = /etc/letsencrypt/live/catplanet.cat/fullchain.pem

# Options and defaults used in the renewal process
[renewalparams]
no_self_upgrade = False
apache_enmod = a2enmod
no_verify_ssl = False
ifaces = None
apache_dismod = a2dismod
register_unsafely_without_email = False
apache_handle_modules = True
uir = None
installer = apache
nginx_ctl = nginx
config_dir = /etc/letsencrypt
text_mode = True
func = <function run at 0x3805c80>
staging = False
dry_run = False
work_dir = /var/lib/letsencrypt
tos = False
init = False
http01_port = 80
duplicate = False
noninteractive_mode = False
key_path = None
nginx = False
nginx_server_root = /etc/nginx
fullchain_path = None
email = None
csr = None
agree_dev_preview = None
redirect = None
verb = run
verbose_count = -3
config_file = None
renew_by_default = False
hsts = False
apache_handle_sites = True
authenticator = apache
domains = catplanet.cat,
rsa_key_size = 2048
apache_challenge_location = /etc/apache2
checkpoints = 1
manual_test_mode = False
apache = True
cert_path = None
webroot_path = ,
reinstall = False
expand = False
strict_permissions = False
apache_server_root = /etc/apache2
account = 7102d106709f6586d54e910aa3d23e07
prepare = False
manual_public_ip_logging_ok = False
chain_path = None
break_my_certs = False
standalone = False
manual = False
server = https://acme-v01.api.letsencrypt.org/directory
standalone_supported_challenges = "tls-sni-01,http-01"
webroot = False
os_packages_only = False
apache_init_script = None
user_agent = None
apache_ctl = None
apache_le_vhost_ext = -le-ssl.conf
debug = False
tls_sni_01_port = 443
logs_dir = /var/log/letsencrypt
apache_vhost_root = /etc/apache2/sites-available
configurator = None
[[webroot_map]]
```

and SSL Labs:
![sslmate](https://cloud.githubusercontent.com/assets/649860/15448448/e1b29ee6-1f16-11e6-92bf-56270f00a10f.jpg)

**Comment by SwartzCr** [comment]
here's my conf after renewal:
```
# renew_before_expiry = 30 days
cert = /etc/letsencrypt/live/catplanet.cat/cert.pem
privkey = /etc/letsencrypt/live/catplanet.cat/privkey.pem
chain = /etc/letsencrypt/live/catplanet.cat/chain.pem
fullchain = /etc/letsencrypt/live/catplanet.cat/fullchain.pem

# Options and defaults used in the renewal process
[renewalparams]
installer = apache
authenticator = apache
account = 7102d106709f6586d54e910aa3d23e07
```

**Comment by bmw** [comment]
This is very odd. `renew_before_expiry` has always been there, however, we recently started adding a comment showing that this value can be set in the config file. If the value is not set in a config file, it defaults to 30 days.

Any thoughts @schoen?

**Comment by pde** [comment]
@SwartzCr any chance the clock on your server was wrong?

**Comment by SwartzCr** [comment]
No it seems pretty accurate. It's a different time zone than my local machine but otherwise is the same day/month

**Comment by pde** [comment]
The relevant logic is here: https://github.com/certbot/certbot/blob/master/certbot/main.py#L148

@SwartzCr if this is reproducible, time to start debugging by dropping an `ipdb.set_trace()` call [here](https://github.com/certbot/certbot/blob/master/certbot/storage.py#L674) or [here](https://github.com/certbot/certbot/blob/master/certbot/renewal.py#L194)?

**Comment by SwartzCr** [comment]
fwiw here's the log from that attempted renewal
```
2016-05-21 12:36:28,063:DEBUG:certbot.main:Root logging level set at 30
2016-05-21 12:36:28,063:INFO:certbot.main:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2016-05-21 12:36:28,063:DEBUG:certbot.main:certbot version: 0.7.0.dev0
2016-05-21 12:36:28,064:DEBUG:certbot.main:Arguments: ['--apache', '-t', '-d', 'catplanet.cat']
2016-05-21 12:36:28,064:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#nginx,PluginEntryPoint#standalone,PluginEntryPoint#manual,PluginEntryPoint#webroot,PluginEntryPoint#apache,PluginEntryPoint#null)
2016-05-21 12:36:28,064:DEBUG:certbot.plugins.selection:Requested authenticator apache and installer apache
2016-05-21 12:36:28,531:DEBUG:certbot.plugins.selection:Single candidate plugin: * apache
Description: Apache Web Server - Alpha
Interfaces: IAuthenticator, IInstaller, IPlugin
Entry point: apache = certbot_apache.configurator:ApacheConfigurator
Initialized: <certbot_apache.configurator.ApacheConfigurator object at 0x12c73d0>
Prep: True
2016-05-21 12:36:28,532:DEBUG:certbot.plugins.selection:Selected authenticator <certbot_apache.configurator.ApacheConfigurator object at 0x12c73d0> and installer <certbot_apache.configurator.ApacheConfigurator object at 0x12c73d0>
2016-05-21 12:36:28,560:DEBUG:certbot.main:Picked account: <Account(7102d106709f6586d54e910aa3d23e07)>
2016-05-21 12:36:28,562:DEBUG:root:Sending GET request to https://acme-v01.api.letsencrypt.org/directory. args: (), kwargs: {}
2016-05-21 12:36:28,565:INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
2016-05-21 12:36:29,017:DEBUG:requests.packages.urllib3.connectionpool:"GET /directory HTTP/1.1" 200 280
2016-05-21 12:36:29,018:DEBUG:root:Received <Response [200]>. Headers: {'Content-Length': '280', 'Expires': 'Sat, 21 May 2016 12:35:01 GMT', 'Strict-Transport-Security': 'max-age=604800', 'Server': 'nginx', 'Connection': 'keep-alive', 'Pragma': 'no-cache', 'Cache-Control': 'max-age=0, no-cache, no-store', 'Date': 'Sat, 21 May 2016 12:35:01 GMT', 'X-Frame-Options': 'DENY', 'Content-Type': 'application/json', 'Replay-Nonce': 'FYfy8vpOxw_nMpSMV3HLDipDtLFCvmU8o3d3fNN6qbM'}. Content: '{\n  "new-authz": "https://acme-v01.api.letsencrypt.org/acme/new-authz",\n  "new-cert": "https://acme-v01.api.letsencrypt.org/acme/new-cert",\n  "new-reg": "https://acme-v01.api.letsencrypt.org/acme/new-reg",\n  "revoke-cert": "https://acme-v01.api.letsencrypt.org/acme/revoke-cert"\n}'
2016-05-21 12:36:29,018:DEBUG:acme.client:Received response <Response [200]> (headers: {'Content-Length': '280', 'Expires': 'Sat, 21 May 2016 12:35:01 GMT', 'Strict-Transport-Security': 'max-age=604800', 'Server': 'nginx', 'Connection': 'keep-alive', 'Pragma': 'no-cache', 'Cache-Control': 'max-age=0, no-cache, no-store', 'Date': 'Sat, 21 May 2016 12:35:01 GMT', 'X-Frame-Options': 'DENY', 'Content-Type': 'application/json', 'Replay-Nonce': 'FYfy8vpOxw_nMpSMV3HLDipDtLFCvmU8o3d3fNN6qbM'}): '{\n  "new-authz": "https://acme-v01.api.letsencrypt.org/acme/new-authz",\n  "new-cert": "https://acme-v01.api.letsencrypt.org/acme/new-cert",\n  "new-reg": "https://acme-v01.api.letsencrypt.org/acme/new-reg",\n  "revoke-cert": "https://acme-v01.api.letsencrypt.org/acme/revoke-cert"\n}'
2016-05-21 12:36:29,031:INFO:certbot.renewal:Cert not yet due for renewal
2016-05-21 12:36:44,651:INFO:certbot_apache.configurator:Deploying Certificate to VirtualHost /etc/apache2/sites-available/default-le-ssl.conf
2016-05-21 12:36:44,651:DEBUG:certbot_apache.configurator:Apache version is 2.2.22
2016-05-21 12:36:44,670:DEBUG:certbot.reverter:Creating backup of /etc/apache2/sites-available/default-le-ssl.conf
2016-05-21 12:36:47,387:INFO:certbot_apache.configurator:Enabled Apache rewrite module
2016-05-21 12:36:47,486:DEBUG:certbot.reverter:Creating backup of /etc/apache2/sites-available/default
2016-05-21 12:36:47,502:INFO:certbot_apache.configurator:Redirecting vhost in /etc/apache2/sites-available/default to ssl vhost in /etc/apache2/sites-available/default-le-ssl.conf
```

**Comment by SwartzCr** [comment]
```
-rw-r--r-- 1 root root 1793 Feb 12 01:10 cert1.pem
-rw-r--r-- 1 root root 1793 May 21 12:32 cert2.pem
-rw-r--r-- 1 root root 1793 May 21 12:45 cert3.pem
-rw-r--r-- 1 root root 1675 Feb 12 01:10 chain1.pem
-rw-r--r-- 1 root root 1647 May 21 12:32 chain2.pem
-rw-r--r-- 1 root root 1647 May 21 12:45 chain3.pem
-rw-r--r-- 1 root root 3468 Feb 12 01:10 fullchain1.pem
-rw-r--r-- 1 root root 3440 May 21 12:32 fullchain2.pem
-rw-r--r-- 1 root root 3440 May 21 12:45 fullchain3.pem
-rw-r--r-- 1 root root 1704 Feb 12 01:10 privkey1.pem
-rw-r--r-- 1 root root 1704 May 21 12:32 privkey2.pem
-rw-r--r-- 1 root root 1704 May 21 12:45 privkey3.pem
```
and
cert3
```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            03:b8:ea:ed:44:47:8c:58:af:78:49:7c:00:ef:1a:08:a5:3e
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=Let's Encrypt, CN=Let's Encrypt Authority X3
        Validity
            Not Before: May 21 11:44:00 2016 GMT
            Not After : Aug 19 11:44:00 2016 GMT
        Subject: CN=catplanet.cat
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:af:2d:c3:b8:bf:78:ed:24:d5:75:7e:01:df:a6:
                    57:b8:23:8e:c3:3a:be:8f:dd:98:23:f7:67:02:43:
                    38:b7:22:87:cb:bc:05:61:a5:19:3d:50:74:e1:f1:
                    ed:7a:83:8f:91:26:8d:60:fa:b6:d4:96:ba:33:0e:
                    e2:5d:c2:49:78:84:73:f3:80:5d:b9:e0:11:01:a3:
                    2a:bd:85:2a:90:41:10:cb:3a:dd:31:4f:73:a6:5c:
                    95:a0:33:26:93:b2:04:3c:a4:06:71:e1:9b:ac:37:
                    f6:41:53:8a:22:2b:1a:a9:e2:f9:e3:64:df:52:92:
                    59:13:55:59:57:86:72:90:8c:d8:02:bb:8a:c9:b6:
                    e8:33:3f:fb:b7:d6:c7:37:57:4a:ab:ce:8e:1b:59:
                    4d:da:71:a3:e1:8a:5d:56:dc:f2:55:b9:b6:c2:31:
                    8c:5f:99:22:b0:5f:0e:4b:2c:ef:59:a2:c5:80:55:
                    eb:d7:13:a0:f3:23:39:de:47:a3:05:31:a4:e1:d9:
                    0b:5f:19:96:3c:53:5f:ec:31:ec:3a:42:39:0e:8f:
                    b7:46:77:4b:7a:e6:89:e6:5a:16:fe:b8:d4:c2:92:
                    10:cd:ff:de:f4:ba:71:ad:ae:20:ab:d5:6a:b4:ab:
                    50:d3:dd:e4:92:ee:7f:34:5a:2d:47:15:04:96:0b:
                    02:6f
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication, TLS Web Client Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier: 
                6B:A5:1D:0C:2F:1B:EA:78:FD:35:CE:4D:D1:66:96:96:6D:A1:FA:BC
            X509v3 Authority Key Identifier: 
                keyid:A8:4A:6A:63:04:7D:DD:BA:E6:D1:39:B7:A6:45:65:EF:F3:A8:EC:A1

            Authority Information Access: 
                OCSP - URI:http://ocsp.int-x3.letsencrypt.org/
                CA Issuers - URI:http://cert.int-x3.letsencrypt.org/

            X509v3 Subject Alternative Name: 
                DNS:catplanet.cat
            X509v3 Certificate Policies: 
                Policy: 2.23.140.1.2.1
                Policy: 1.3.6.1.4.1.44947.1.1.1
                  CPS: http://cps.letsencrypt.org
                  User Notice:
                    Explicit Text: This Certificate may only be relied upon by Relying Parties and only in accordance with the Certificate Policy found at https://letsencrypt.org/repository/

    Signature Algorithm: sha256WithRSAEncryption
         62:17:63:ae:ee:c4:96:ec:b7:ca:6f:56:58:0d:34:ea:e8:b8:
         95:57:e9:d8:d2:07:6e:07:0b:92:0b:6a:1b:05:7e:50:a4:21:
         53:a3:ad:29:a6:32:6f:ac:9d:01:14:87:ae:1b:cc:a4:4e:4e:
         d8:70:e2:ef:dd:eb:52:41:27:2b:b3:9e:09:5e:ae:77:f9:5a:
         89:9a:f1:ff:31:57:04:3a:63:b1:04:9f:cf:2c:66:a8:c8:42:
         ec:f8:c6:d6:43:89:fa:d3:c2:04:6f:43:66:91:fb:aa:cc:6b:
         d0:d8:5e:12:d7:fe:7e:c1:aa:20:67:ad:26:e3:91:40:bb:3e:
         f5:72:55:53:32:6a:3d:fb:5a:8b:ff:d9:b1:c8:c0:1f:84:4c:
         2b:8e:fd:26:e0:c3:b5:9c:cf:c0:b7:9c:7a:51:56:6b:30:21:
         61:24:3e:c3:45:a8:f7:ca:de:6e:c2:be:c3:6d:58:da:08:7d:
         a7:0a:e4:f2:11:ff:4a:9b:6d:1e:49:f2:4f:5a:18:3b:69:af:
         4a:43:0a:a6:52:6e:1c:aa:20:86:56:aa:33:52:d7:68:c9:7b:
         22:c5:c4:14:7d:0b:80:bd:6b:94:4d:aa:15:88:99:8f:d8:07:
         8e:25:5b:a2:71:19:f0:92:fa:c4:02:a8:76:5c:43:35:97:24:
         04:ee:ee:e1
```
cert2
```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            03:62:ed:0d:bf:77:ad:89:63:18:3a:54:ac:ad:02:1e:1f:22
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=Let's Encrypt, CN=Let's Encrypt Authority X3
        Validity
            Not Before: May 21 11:31:00 2016 GMT
            Not After : Aug 19 11:31:00 2016 GMT
        Subject: CN=catplanet.cat
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:ba:07:9e:2c:75:c8:6a:cb:55:34:14:5c:c6:c6:
                    30:97:98:6e:d9:57:fa:12:92:38:f2:60:1b:c0:d0:
                    2f:7f:a4:8e:1d:c4:a0:de:d1:ec:dd:b1:16:59:aa:
                    bb:4b:2c:c9:fd:da:33:1b:45:c7:de:af:f5:61:2c:
                    7a:a1:23:12:5d:b9:17:37:42:69:96:da:65:d4:80:
                    c6:72:92:26:ac:ed:d9:a3:0b:57:e6:9a:a3:2f:01:
                    52:ca:80:1e:bf:36:d0:86:3d:ac:70:1b:0c:b8:b9:
                    26:02:c0:62:25:d0:7a:38:9f:7b:74:d1:e3:19:e3:
                    51:65:30:e5:b4:19:3a:2f:3f:61:24:a5:8f:70:64:
                    93:38:fa:c9:a5:d3:a7:9b:2c:57:da:92:4d:c9:90:
                    f2:b9:05:3c:a8:28:6b:ab:2e:b0:dd:6f:98:6c:ae:
                    d5:95:5e:b6:83:33:8a:83:18:5d:f9:e5:36:db:0c:
                    a7:08:ed:f2:cd:1e:54:ba:d5:30:d4:8d:bb:ac:df:
                    79:bc:08:01:f6:2d:03:5a:e0:85:85:38:e1:92:02:
                    76:96:17:d9:2d:1c:2e:16:3b:1c:70:6f:da:84:d6:
                    3a:50:8f:64:63:4b:51:f9:c0:70:d4:21:a0:eb:50:
                    ab:14:7c:57:9f:f4:1c:12:e6:7c:2e:30:4e:3b:eb:
                    cb:19
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication, TLS Web Client Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier: 
                A4:9D:F7:1E:86:08:CC:99:F9:0C:F0:06:F6:A2:03:3B:A7:A5:67:A3
            X509v3 Authority Key Identifier: 
                keyid:A8:4A:6A:63:04:7D:DD:BA:E6:D1:39:B7:A6:45:65:EF:F3:A8:EC:A1

            Authority Information Access: 
                OCSP - URI:http://ocsp.int-x3.letsencrypt.org/
                CA Issuers - URI:http://cert.int-x3.letsencrypt.org/

            X509v3 Subject Alternative Name: 
                DNS:catplanet.cat
            X509v3 Certificate Policies: 
                Policy: 2.23.140.1.2.1
                Policy: 1.3.6.1.4.1.44947.1.1.1
                  CPS: http://cps.letsencrypt.org
                  User Notice:
                    Explicit Text: This Certificate may only be relied upon by Relying Parties and only in accordance with the Certificate Policy found at https://letsencrypt.org/repository/

    Signature Algorithm: sha256WithRSAEncryption
         29:c1:0c:f0:a4:01:f1:e4:06:fa:eb:49:3b:06:ac:3a:97:11:
         50:85:c2:c1:39:9c:20:9d:4a:7d:26:c2:11:24:e7:3a:8b:41:
         21:98:78:dd:91:4a:e5:62:e3:47:9b:23:d0:65:08:a6:65:39:
         4f:e7:5e:63:ac:80:7c:d1:c1:0d:21:13:a6:e1:31:05:32:dd:
         05:37:8c:57:bf:6b:68:dd:d6:23:32:66:1a:b3:b7:dd:93:46:
         d7:fe:ec:cc:38:d5:c7:ce:e4:84:2e:99:34:62:2b:b2:96:31:
         a2:64:78:7e:03:66:1d:2c:bf:a3:37:92:d6:90:04:ac:e7:d8:
         e6:21:c6:78:1d:2f:9a:8c:32:ef:fc:ee:99:a2:d9:9f:a4:12:
         43:6e:eb:af:20:44:e0:ee:ad:3a:b6:47:0c:b4:ec:b5:dd:99:
         08:3e:82:7f:86:0f:ab:5f:2a:f9:02:2b:f6:be:b9:f9:fd:a8:
         3a:be:cc:8c:b1:58:e6:f2:3b:a1:c5:18:23:11:f6:5e:7b:ed:
         bb:db:28:ff:39:74:5b:4f:74:1b:3a:bb:7a:18:60:d8:4d:72:
         2d:95:2e:ad:71:c0:b6:83:80:6d:d5:90:52:5a:f2:57:1c:40:
         16:c6:bf:4d:68:77:0a:2f:a2:e8:55:cb:53:14:bd:6d:55:4e:
         5a:b1:05:07
```
cert1
```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            01:22:dd:c6:b5:72:8b:f2:ea:dc:91:e2:da:b1:69:f9:da:fa
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=Let's Encrypt, CN=Let's Encrypt Authority X1
        Validity
            Not Before: Feb 12 00:11:00 2016 GMT
            Not After : May 12 00:11:00 2016 GMT
        Subject: CN=catplanet.cat
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:d0:eb:08:bf:9d:3b:4c:4b:35:41:65:03:1f:17:
                    04:31:be:aa:bb:58:c4:09:38:e0:7a:a4:14:fa:c9:
                    ac:b5:99:8b:50:6a:93:a6:f4:e9:da:bc:8a:49:0c:
                    cb:ae:d6:c0:0c:27:8c:41:ea:e1:0c:16:0b:3d:ca:
                    99:34:cd:59:03:05:7f:bb:07:a3:6f:07:18:d9:0e:
                    dc:24:c7:67:83:0b:fe:1c:48:2b:14:ab:89:b6:d0:
                    53:50:e5:7f:8b:11:4d:73:e5:c2:1e:31:96:0e:99:
                    86:ea:94:85:92:d2:58:ca:03:27:0d:ee:61:8d:56:
                    bd:3a:1b:e6:00:80:1a:60:ac:38:f6:31:41:79:57:
                    28:4c:45:56:fe:dd:51:a5:27:49:1b:a8:81:08:84:
                    ef:29:7c:68:35:2b:aa:92:f3:d0:a4:ec:a6:c2:2c:
                    96:74:46:ce:e7:2e:6e:d3:92:2a:52:27:ab:71:d7:
                    33:34:34:9a:07:22:d1:ce:0d:2d:8c:44:a8:d9:3d:
                    d0:5a:80:12:41:88:66:dc:da:06:21:b8:66:79:f3:
                    f0:b7:9d:fb:a0:69:fd:72:61:b4:9b:02:bb:ea:99:
                    79:46:21:b6:1f:e1:26:20:cd:4b:9a:98:58:5b:e6:
                    47:fd:fd:7e:4f:b4:8f:c3:e9:17:41:b5:5d:d5:65:
                    39:c7
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication, TLS Web Client Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier: 
                F8:54:3F:B0:9C:1A:20:BE:32:D0:69:22:18:06:BF:FB:26:B4:8F:46
            X509v3 Authority Key Identifier: 
                keyid:A8:4A:6A:63:04:7D:DD:BA:E6:D1:39:B7:A6:45:65:EF:F3:A8:EC:A1

            Authority Information Access: 
                OCSP - URI:http://ocsp.int-x1.letsencrypt.org/
                CA Issuers - URI:http://cert.int-x1.letsencrypt.org/

            X509v3 Subject Alternative Name: 
                DNS:catplanet.cat
            X509v3 Certificate Policies: 
                Policy: 2.23.140.1.2.1
                Policy: 1.3.6.1.4.1.44947.1.1.1
                  CPS: http://cps.letsencrypt.org
                  User Notice:
                    Explicit Text: This Certificate may only be relied upon by Relying Parties and only in accordance with the Certificate Policy found at https://letsencrypt.org/repository/

    Signature Algorithm: sha256WithRSAEncryption
         4b:68:bf:a4:c5:13:44:0f:b0:31:e0:5c:fd:d6:79:f3:90:ee:
         73:06:a0:87:d7:cc:c6:e5:8c:ce:79:bd:54:cd:c9:b3:e9:e5:
         ab:1d:ae:7a:88:6e:47:f8:88:8e:12:26:5a:61:99:3b:5c:16:
         9b:19:4b:fb:4f:14:ab:4d:ee:bd:ec:d8:52:b9:4e:06:59:6f:
         80:cc:ee:cb:7f:f4:0a:95:0d:33:4f:a9:18:ea:89:fd:59:3a:
         55:79:51:09:22:09:6d:d1:51:8e:0a:cd:26:88:04:67:45:68:
         bd:c0:d5:f5:33:a5:a2:97:d4:63:aa:75:fa:46:02:83:9b:f1:
         03:a8:b4:58:a9:13:66:99:ac:f2:e9:19:47:63:17:65:45:bb:
         68:c3:68:ab:7e:88:e5:55:32:1e:6e:ac:27:07:f9:18:4e:94:
         e0:ce:fe:a2:e4:86:f6:9d:41:ee:6e:d0:07:05:7e:2c:62:0c:
         83:6e:2e:a3:a0:1a:b8:99:b6:80:2b:71:4d:de:73:9d:9a:44:
         42:37:ad:35:f0:26:77:e9:c6:45:20:71:fb:f4:58:48:cb:52:
         0f:52:9a:48:03:5a:c4:e6:d3:0f:85:ea:f9:0b:47:10:48:d3:
         a2:6a:75:88:ab:ca:2e:e8:76:6d:83:d2:10:fc:ed:9b:79:08:
         9e:d4:0c:59
```

**Comment by SwartzCr** [comment]
```
-rw-r--r--  1 root root 46556 May 21 12:45 letsencrypt.log  
-rw-r--r--  1 root root  4048 May 21 12:36 letsencrypt.log.1
-rw-r--r--  1 root root 38738 Nov 20  2015 letsencrypt.log.10
-rw-r--r--  1 root root 49854 May 21 12:32 letsencrypt.log.2  
-rw-r--r--  1 root root 43066 Feb 25 15:36 letsencrypt.log.3
-rw-r--r--  1 root root 46777 Feb 12 02:09 letsencrypt.log.4
-rw-r--r--  1 root root  1569 Feb 12 01:06 letsencrypt.log.5  
-rw-r--r--  1 root root 39060 Nov 20  2015 letsencrypt.log.6
-rw-r--r--  1 root root 38835 Nov 20  2015 letsencrypt.log.7
-rw-r--r--  1 root root 15171 Nov 20  2015 letsencrypt.log.8
-rw-r--r--  1 root root 18770 Nov 20  2015 letsencrypt.log.9
```
log.2: http://paste.debian.net/739565/
log.1: http://paste.debian.net/739566/
log: http://paste.debian.net/739567

**Comment by schoen** [comment]
Thanks for providing all of this data, @SwartzCr.

I think there was an issue with renewal-without-deployment (that is, without updating the symlinks), which led the client to conclude that a subsequent renewal was unnecessary because there was already an up-to-date cert on the disk. But the message displayed in this case may not have correctly explained this because I think it wouldn't have differentiated between "do you want me to update the symlinks to deploy your existing renewed cert?" and "do you want me to reinstall your existing cert?".

If my interpretation is right, this is a slightly hard-to-trigger (albeit quite confusing) UI bug, rather than an actual outright failure of the renewal logic.

I can try to map out the sequence of events from your logs in more detail to confirm this interpretatoin.

**Comment by pde** [comment]
@schoen if this is a real bug let's fix it in 0.11.0. If not, let's close this!

**Comment by bmw** [comment]
I just played with this for a bit. The issue as hypothesized by Seth of having a pending deployment has been resolved. First, Certbot does not currently respect values for `autodeploy` or `deploy_before_expiry` in renewal conf files. Some of the logic is still there with `autodeployment_is_enabled` and `should_autodeploy`, but it's not called anywhere (other than in tests). I added #4315 to track removing this code (we always have our git history if we want it in the future).

With this in mind, I manually created the situation of having an expired certificate and an undeployed certificate. When running Certbot, I got the message: "Found a new cert /archive/ that was not linked to in /live/; fixing...". It then automatically deployed the lineage and offered to reinstall it. This logic was added in #3560 after this issue was opened. Because the only explanation we have for this is a fixed bug and no one else has reported it, I'm closing the issue. Looking at the log above, I can see Noah was running a development version of Certbot. Perhaps we temporarily had a broken version in our tree.

If anyone can repo, please let me know and I'll reopen.
